test/junit/scala/tools/nsc/interpreter/CompletionTest.scala

```diff
@@ -356,4 +356,24 @@ object Test2 {
     assertEquals(expected.sorted.mkString(" "), actual.toSeq.distinct.sorted.mkString(" "))
   }
 
+  @Test
+  def ignoreAlias(): Unit = {
+    val (completer, _, _) = interpretLines(
+      """class Foo(i: Int) { def this(s: String) = this(s.toInt) }""",
+      """type Bar = Foo"""
+      )
+    val candidates1       = completer.complete("new Foo").candidates
+    val actual            = candidates1.map(_.declString).map(_.apply()).mkString(" ")
+    // def <init>(i: Int): Foo   def <init>(s: String): Foo
+    assertEquals(2, candidates1.size)
+    assertEquals("def <init>(i: Int): Foo def <init>(s: String): Foo", actual)
+
+
+    val candidates2 = completer.complete("new Bar").candidates
+    assertEquals(1, candidates2.size)
+    val actual2 = candidates2.map(_.declString).map(_.apply()).mkString(" ")
+    // This seems meaningless, but at present I don't know if it's worth keeping, so I'll keep the alias as well.
+    assertEquals("type Bar = Bar", actual2)
```

Comment:
So nothing is fixed?  That's not the

```
type Bar = Bar
def <init>(i: Int): Foo 
def <init>(s: String): Foo
```

you mentioned in the PR description.

Reply:
No, unit test is to ensure that the previous logic is not modified,  the individual candidates are still the same as before, and only the display behavior is added in `Reader.scala`. 
Because the candidates are stored separately `result.candidates.filter(c => c.name == parsedLineWord || c.alias.fold(false)(a => a == parsedLineWord))`  

> As I commented in PR, I'm not sure about `new Bar`, whether  `type bar = bar` is necessary. If we remove the `type Bar = Bar` directly, the previous test will be affected

Reply:
Can we write a test that asserts on the final result then?  Otherwise there's no proof of the fix, as I see it.

Reply:
Yes, I'll try it

Reply:
I advanced the changes to the match phase of the `repl` module.  So this makes it easier to write tests.